pages/coupon/create.js

```diff
@@ -0,0 +1,257 @@
+import CommerceLayout from '@components/common/CommerceLayout';
+import PostButton from '@components/input/button';
+import { GET_CUSTOM, POST } from '@apis/defaultApi';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import useInput from '@hooks/useInput';
+
+export default function CouponCreate() {
+  const router = useRouter();
+  const [name, onChangeName] = useInput('');
+  const [userId, onChangeUserId] = useInput(4);
+  const [type, onChangeType] = useInput(1);
+  const [discountValue, onChangeDiscountValue] = useInput(0);
+  const [validAt, onChangeValidAt] = useInput(Date);
+  const [validTime, onChangeValidTime] = useInput();
+  const [expiresAt, onChangeExpiresAt] = useInput(Date);
+  const [expireTime, onChangeExpireTime] = useInput();
+  const [maxDiscountAmount, onChangeMaxDiscountAmount] = useInput(0);
+  const [minPaymentAmount, onChangeMinPaymentAmount] = useInput(0);
+  const [detail, onChangeDetail] = useInput('');
+  const [cnt, onChangeCnt] = useInput();
+
+  const [role, setRole] = useState('');
+
+  // TODO:
+  // setUserId(현재로그인되어있는userId-세션,쿠키 등에서 얻어올 것임);
+
+  useEffect(() => {
+    GET_CUSTOM(`/user/role`, { userId }).then((res) => {
+      if (res.message === 'ROLE_USER') {
+        setRole('USER');
+      }
+      if (res.message === 'ROLE_SELLER') {
+        setRole('SELLER');
+      }
+    });
+  }, []);
+
+  if (role === 'USER') {
+    return (
+      <CommerceLayout>
+        <p>
+          해당 사용자는 판매자가 아니라 쿠폰을 등록할 수 없습니다. 판매자로
+          로그인 후에 사용해주세요.
+        </p>
+      </CommerceLayout>
+    );
+  }
+
+  function submitFormHandler(e) {
+    e.preventDefault();
+
+    const reqBody = {
+      name: name,
+      userId: userId,
+      type: type,
+      discountValue: discountValue,
+      validAt: validAt + 'T' + validTime,
+      expiresAt: expiresAt + 'T' + expireTime,
+      maxDiscountAmount: maxDiscountAmount,
+      minPaymentAmount: minPaymentAmount,
+      detail: detail,
+      cnt: cnt,
+    };
+
+    POST(`/coupon/create`, reqBody)
+      .then((res) => {
+        console.log(res);
+
+        alert(
+          res.sellerName +
+            ' 판매자님의 ' +
+            res.type +
+            " 유형의 '" +
+            res.couponName +
+            "' 쿠폰이 \n" +
+            res.cnt +
+            ' 장 발급 완료 되었습니다.',
+        );
+        router.push(`./list`);
+      })
+      .catch(function (error) {
+        console.log(error);
+        return {};
+      });
+  }
+
+  // TODO: Dropdown select 수행 관련 함수
+  // function changeTypeSelect() {
+  //   var typeSelect = querySelector('#type-select');
+  //   var selectValue = typeSelect.options[typeSelect.selectedIndex].value;
+  //   var selectText = typeSelect.options[typeSelect.selectedIndex].text;
+
+  //   console.log(selectValue);
+  //   console.log(selectText);
+  // }
+
+  return (
+    <CommerceLayout>
+      <SiteHead title="쿠폰 등록" />
+      <Divider />
+      <Heading title="쿠폰 등록" type="h1"></Heading>
```

Comment:
```suggestion
      <Heading title="쿠폰 등록" type="h1"/>
```

가독성은 이게 더 좋아보입니다! 내부에 컴포넌트 들어가는 게 아닌거는 그냥 앞에서 다 닫아주세요(다른것두요!)

Reply:
셀러 스토어 관련된 모든 페이지는 /seller 부터 시작한다고 생각해야겠네욤

seller 쿠폰 목록 조회할때 coupon/seller/list 가 아닌 seller/coupon/list 로 하는 이유는 스프링 시큐리티를 사용해주기 위해서 인가요? 

아래 comment 의 res.sellerName 수정은 mono-be 가 수정되어야 해서 추후에 수정할게요

Reply:
@daeunchung 

우선 우리 프로젝트는 관리자 페이지 메인이 /seller/index 으로 되어있어요.

판매자가 사용하는 입장에서 생각해 봅시당!

쿠폰관리 페이지는 판매자 페이지 내부에서 쿠폰관리 페이지 버튼을 눌러서 들어갈거죠?
쿠폰관리 페이지는 관리자 페이지의 내부라고 생각할 수 있을 거 같아요.

계층관계를 생각할때 사용자 페이지 메인 / 판매자 메인 이 제일 높은 계층이라고 생각해요.
쿠폰 페이지는 저 두 페이지의 내부에 있는 페이지라고 생각합니다.

Reply:
자세한 설명 감사합니다
그러면 사용자 coupon list 페이지는 /coupon/list 로,
판매자가 쿠폰을 생성하는 페이지는 /seller/coupon/create 로 
사용자가 쿠폰을 사용하는 페이지는 /coupon/use
판매자가 쿠폰을 사용자에게 배정하는 페이지는 /seller/coupon/assign 으로 하면 될까요?

Reply:
네넹~ 좋아요~